org-roam-graph.el

```diff
@@ -122,21 +123,18 @@ set WHERE to true if WHERE query already exists."
       (push match res))
     (nreverse res)))
 
-(defun org-roam-graph--build ()
-  "Build the graph output string.
+(defun org-roam-graph--build (node-query)
+  "Build the graph output string for NODE-QUERY.
 The Org-roam database titles table is read, to obtain the list of titles.
 The file-links table is then read to obtain all directed links, and formatted
 into a digraph."
   (org-roam-db--ensure-built)
   (org-roam--with-temp-buffer
-    (let* ((node-query `[:select [file titles]
-                                 :from titles
-                                 ,@(org-roam-graph--expand-matcher 'file t)])
-           (nodes (org-roam-db-query node-query))
-           (edges-query `[:select :distinct [file-to file-from]
-                                  :from file-links
-                                  ,@(org-roam-graph--expand-matcher 'file-to t)
-                                  ,@(org-roam-graph--expand-matcher 'file-from t t)])
+    (let* ((nodes (org-roam-db-query node-query))
```

Comment:
Alternatively we might accept just the `:where` expression, instead of passing in the full query. (Since the rest of this function depends on the "file" and "titles" columns being returned in the correct order.

Reply:
The benefit of the current approach is that we could pass in e.g.
```elisp
[:select [file titles]
 :from file-links
 :join titles :on (or (= file file-from)
                      (= file file-to))
 ,@(org-roam-graph--expand-matcher 'file t)]
```
to display only notes that have any connections at all.